lsp-tests/Test.hs

```diff
@@ -25,21 +25,30 @@ hoverTests = testGroup "Hover tests"
   , expectFail                                $ testHover "Hover over nothing"                                        "mini.l4" (Position 1 0)   (mkRange 3 0 3 8)    "This block maps variable Business to WordNet definion \"business_1_N\""
   ]
 
-typeCheckerTests :: TestTree 
-typeCheckerTests = testGroup "Type Error tests"  
+hoverTypeInfoTests :: TestTree
+hoverTypeInfoTests = testGroup "Hover type tests"
+  [ testHover "mini.l4 type info for lexicon DetractsFromDignity"   "mini.l4" (Position 4 4)   (mkRange 4 0 4 66)   "OkT"
+  , testHover "cr.l4 type info for 'class Business {'"              "cr.l4"   (Position 21 10) (mkRange 21 0 24 1)  "OkT"
+  , testHover "cr.l4 type info for var decl AssociatedWith"         "cr.l4"   (Position 38 12) (mkRange 38 0 38 64) "OkT"
+  , testHover "cr.l4 type info for rule r1a"                        "cr.l4"   (Position 61 7)  (mkRange 61 0 64 29) "ClassT (ClsNm {stringOfClassName = \"Boolean\"})"
+  , testHover "cr.l4 type info for assert"                          "cr.l4"   (Position 103 3) (mkRange 103 0 103 83) "ClassT (ClsNm {stringOfClassName = \"Boolean\"})"
+  ]
```

Comment:
Could we have a test that doesn't return Boolean, but rather covers some sub-expression of another type?